bug: SafePath() fails to block path traversal on Windows — vulnerability

## Summary

`SafePath()` in `internal/web/safepath.go` fails to reject absolute path traversal on Windows. The existing unit test `TestSafePath/traversal_absolute` **already fails** on Windows, confirming this bug:

```
safepath_test.go:26: SafePath("/tmp/state", "/etc/passwd") error = <nil>, wantErr true
```

This is a security concern because `SafePath` is designed to prevent directory traversal attacks, but it silently allows them on Windows.

---

## Environment

- **OS:** Windows 10 / 11 (amd64)
- **Go:** 1.26.0
- **PinchTab:** v0.7.6 (commit `82ab9a1`)

---

## Reproduction

```bash
go test -v -run TestSafePath/traversal_absolute ./internal/web/
```

**Output on Windows:**

```
=== RUN   TestSafePath
=== RUN   TestSafePath/traversal_absolute
    safepath_test.go:26: SafePath("/tmp/state", "/etc/passwd") error = <nil>, wantErr true
--- FAIL: TestSafePath/traversal_absolute (0.00s)
FAIL    github.com/pinchtab/pinchtab/internal/web    0.373s
```

---

## Root Cause

On Windows, `filepath.IsAbs("/etc/passwd")` returns **`false`** because Unix-style absolute paths (no drive letter like `C:\`) are not considered absolute.

This causes `SafePath()` to silently take the wrong code branch in `internal/web/safepath.go`:

```go
// Current code — lines 18-23
if filepath.IsAbs(userPath) {
    resolved = filepath.Clean(userPath)          // ← intended path for absolute inputs
} else {
    resolved = filepath.Clean(filepath.Join(absBase, userPath))  // ← wrong branch taken on Windows
}
```

**Step-by-step breakdown on Windows:**

| Step | Unix (correct ✅) | Windows (broken ❌) |
|------|-------------------|----------------------|
| Input | `"/etc/passwd"` | `"/etc/passwd"` |
| `filepath.IsAbs("/etc/passwd")` | `true` | **`false`** |
| Code branch | `filepath.Clean("/etc/passwd")` → `/etc/passwd` | `filepath.Join("C:\tmp\state", "/etc/passwd")` → **`C:\tmp\state\etc\passwd`** |
| Prefix check against `C:\tmp\state\` | `false` → **blocked ✅** | `true` → **allowed ❌** |

The attacker-supplied path `/etc/passwd` gets **joined into the base directory** instead of being rejected, making it look like a valid subdirectory.

I verified this with a minimal reproducer:

```go
base     := "/tmp/state"
userPath := "/etc/passwd"

absBase, _ := filepath.Abs(base)
// Windows: absBase = "C:\tmp\state"

isAbs := filepath.IsAbs(userPath)
// Windows: isAbs = false  ← BUG HERE

resolved := filepath.Clean(filepath.Join(absBase, userPath))
// Windows: resolved = "C:\tmp\state\etc\passwd"  ← looks safe but isn't

strings.HasPrefix(resolved, absBase+`\`)
// Windows: true  ← passes the check incorrectly
```

---

## Proposed Fix

Add `filepath.Abs()` after the initial resolution to normalize drive letters on Windows. This ensures a Unix-style path like `\etc\passwd` becomes `C:\etc\passwd`, which correctly fails the prefix check against `C:\tmp\state\`.

```go
func SafePath(base, userPath string) (string, error) {
	absBase, err := filepath.Abs(base)
	if err != nil {
		return "", fmt.Errorf("invalid base path: %w", err)
	}

	var resolved string
	if filepath.IsAbs(userPath) {
		resolved = filepath.Clean(userPath)
	} else {
		resolved = filepath.Clean(filepath.Join(absBase, userPath))
	}

	// NEW: Normalize to true absolute path — fixes Windows drive-letter handling
	// On Windows, filepath.IsAbs("/etc/passwd") is false, causing it to be joined
	// into the base. filepath.Abs() then resolves it correctly with the drive letter.
	resolved, err = filepath.Abs(resolved)
	if err != nil {
		return "", fmt.Errorf("invalid resolved path: %w", err)
	}

	if !strings.HasPrefix(resolved, absBase+string(filepath.Separator)) && resolved != absBase {
		return "", fmt.Errorf("path %q escapes base directory %q", userPath, absBase)
	}

	return resolved, nil
}
```

The test cases should also be extended with Windows-specific traversal paths:

```go
// Existing (already in test file — passes on Unix, fails on Windows):
{"traversal absolute", "/tmp/state", "/etc/passwd", true},

// New cases to add for full Windows coverage:
{"traversal windows other drive", `C:\tmp\state`, `D:\secret\file.txt`, true},
{"traversal backslash root",      `C:\tmp\state`, `\Windows\System32\config`, true},
```

---

## Impact

| | |
|---|---|
| **Severity** | Medium-High (path traversal bypass) |
| **Platform** | Windows only — Unix behavior is correct |
| **Affected component** | `internal/web/safepath.go` |
| **Fix complexity** | One line — adding `filepath.Abs(resolved)` |
| **Tests** | Existing test already catches it — just needs the fix to pass |

The fix is a one-line change. The test already exists and already demonstrates the failure. This is the right time to fix it before `SafePath` is used more widely across handlers.

---

## References

- Affected file: `internal/web/safepath.go`
- Failing test: `internal/web/safepath_test.go:26` (`TestSafePath/traversal_absolute`)
- Go issue context: `filepath.IsAbs` behavior on Windows: https://pkg.go.dev/path/filepath#IsAbs


## Contributors

@YashJadhav21 @Chetnapadhi

--- a/internal/web/safepath.go
+++ b/internal/web/safepath.go
@@ -21,6 +21,11 @@ func SafePath(base, userPath string) (string, error) {
 		resolved = filepath.Clean(filepath.Join(absBase, userPath))
 	}
 
+	resolved, err = filepath.Abs(resolved)
+	if err != nil {
+		return "", fmt.Errorf("invalid resolved path: %w", err)
+	}
+
 	// Ensure resolved path is within or equal to base
 	if !strings.HasPrefix(resolved, absBase+string(filepath.Separator)) && resolved != absBase {
 		return "", fmt.Errorf("path %q escapes base directory %q", userPath, absBase)

